docs/source/plugins.rst

```diff
@@ -168,6 +173,14 @@ plugin:
        within the ``mlflow_test_plugin`` module) to register.
      - `GitRunContext <https://github.com/mlflow/mlflow/blob/branch-1.13/mlflow/tracking/context/git_context.py#L38>`_,
        `DefaultRunContext <https://github.com/mlflow/mlflow/blob/branch-1.13/mlflow/tracking/context/default_context.py#L41>`_
+   * - Plugins for specifying custom experiment id lookup/creation at runtime. The plugin must have a way to create or look up the experiment id.
+     - mlflow.default_experiment_provider
+     - The entry point name is unused. The entry point value (e.g. ``mlflow_test_plugin.default_experiment_provider:PluginDefaultExperimentProvider``) specifies a custom subclass of
+       `mlflow.tracking.default_experiment.abstract_context.DefaultExperimentProvider <https://github.com/mlflow/mlflow/blob/branch-2.2/mlflow/tracking/default_experiment/abstract_context.py#L6>`_
```

Comment:
I just realized that the existing links point to really old release branches. Could we update the implementation guidance references from <branch-x.x> to `master`?

Reply:
If linking to `master`, can we avoid including line numbers as well, since they'll change

Reply:
I can link to the files, but over the years they'll 404. Which, actually might not be a bad thing. Thoughts on updating to `"py:class` type of sphinx refs? That way, if someone changes the class, the docs break and they have to update it. 

Just need to install the plugin as part of the docs ci stage.

Reply:
^ if someone can point me to where the docs are generated in ci, will take care of it.

Reply:
@jmahlik the docs build CI config is here: https://github.com/mlflow/mlflow/blob/master/.circleci/config.yml

Reply:
Any new plugins would just need to be added here: https://github.com/mlflow/mlflow/blob/master/requirements/doc-requirements.txt